test/api/api_test.hpp

```diff
@@ -43,3 +45,15 @@ void print_compare_junction_vectors(std::vector<Junction> const & junctions_expe
                              << junctions_expected_res[i].get_tandem_dup_count() << " == " << junctions_res[i].get_tandem_dup_count() << "\n";
     }
 }
+
+// Helper function for comparing two bamit trees.
+void compare_bamit_trees(std::unique_ptr<bamit::IntervalNode> const & t1, std::unique_ptr<bamit::IntervalNode> const & t2)
```

Comment:
Note that a failing EXPECT command does not stop the test execution and we always walk through each tree completely. In case you don't want to see a huge amount of error messages if two trees differ, this function could return a boolean whether they differ (and the walk can be pruned after the first difference) and then you check the function result with a single EXPECT statement. Another, more aggressive method is using `ASSERT_EQ` which aborts and fails the whole test immediately (I doubt that this is desired as you don't find further errors).